skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala

```diff
@@ -150,7 +150,7 @@ object Producer {
 
           def block(record: ProducerRecordJ[Bytes, Bytes]) = {
 
-            def callbackOf(deferred: Deferred[F, Either[Throwable, RecordMetadataJ]]): Callback = {
+            def callbackOf(promise: Promise[RecordMetadataJ]): Callback = {
```

Comment:
why do you think that this is anyhow faster?

Reply:
I'd recommend keep using `deferred` but call `toFuture` instead `toTry` and deprecate API requiring `ToTry`

Reply:
But how is `ToFuture` better than `ToTry`, don't they both call `unsafeRunSync` in implementation?

Reply:
> But how is ToFuture better than ToTry, don't they both call unsafeRunSync in implementation?

1. less construction requirements
2. `ToFuture` uses `unsafeToFuture`, and truly saying both are ok in this case

Reply:
So if I change `ToTry` to `ToFuture` what do I do with the resulting `Future`. Do I `await` inside the `Callback` or just ignore it? TBH I don't like any of these options and would rather leave `ToTry` then.
btw, what's the reason you are against using `Promise` here? To me it seems cleaner that materialising an effect in any manner

Reply:
> Do I await inside the Callback or just ignore it?

Just ignore it, as executed code is private and you exactly know what is being executed

Reply:
> the reason you are against using Promise here

Because that combination of Promise/Future adds more overhead that you might think of comparing to native, it is also not a cancellable thing.

Reply:
But this particular use case is not cancellable either way, is it? Even if you cancel the resulting effect the producer's callback will still be executed

Reply:
I mean cancelling the "inner" effect resulting from `Deferred#get`